SECP-service/src/main/resources/assets/app/views/portal/filter.html

```diff
@@ -0,0 +1,124 @@
+                <option value="">Select</option>
+                <option value="permission1">Top Secret</option>
+                <option value="permission2">Secret</option>
+                <option value="permission3">Basic</option>
+            </select>
+        </td>
+        <td>
+            <select class="role-select" style="width: 100%" name="roles[]" multiple="multiple">
+                <option value="">Select</option>
+                <option value="role1">General</option>
+                <option value="role2">Developer</option>
+                <option value="role3">Manager</option>
+            </select>
+        </td>
+        <td style="text-align: right"><button data-target="#delete"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+    </tr>
+    <tr>
+        <td>Pheonix</td>
+        <td>
+            <select class="permission-level-select" style="width: 100%" name="permission[]" multiple="multiple">
+                <option value="">Select</option>
+                <option value="permission1">Top Secret</option>
+                <option value="permission2">Secret</option>
+                <option value="permission3">Basic</option>
+            </select>
+        </td>
+        <td>
+            <select class="role-select" style="width: 100%" name="roles[]" multiple="multiple">
+                <option value="">Select</option>
+                <option value="role1">General</option>
+                <option value="role2">Developer</option>
+                <option value="role3">Manager</option>
+            </select>
+        </td>
+        <td style="text-align: right"><button data-target="#delete"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+    </tr>
+    <tr>
+        <td>Onyx</td>
+        <td>
+            <select class="permission-level-select" style="width: 100%" name="permission[]" multiple="multiple">
+                <option value="">Select</option>
+                <option value="permission1">Top Secret</option>
+                <option value="permission2">Secret</option>
+                <option value="permission3">Basic</option>
+            </select>
+        </td>
+        <td>
+            <select class="role-select" style="width: 100%" name="roles[]" multiple="multiple">
+                <option value="">Select</option>
+                <option value="role1">General</option>
+                <option value="role2">Developer</option>
+                <option value="role3">Manager</option>
+            </select>
+        </td>
+        <td style="text-align: right"><button data-target="#delete"><i class="fa fa-trash red" aria-hidden="true"></i></button></td>
+    </tr>
+    </tbody>
+</table>
+</body>
+
+<!-- Modal -->
+<div class="modal fade" id="myModal" role="dialog">
+    <div class="modal-dialog">
+
+        <!-- Modal content-->
+        <form class="modal-content">
```

Comment:
I see no form verification here. This can be done on the html side. Just verify that if the user clicks on save and no input is provided, some red bar shows. You need the required keyword.

Reply:
✔️